src/smexperiments/experiment.py

```diff
@@ -223,3 +224,31 @@ def create_trial(self, trial_name=None, trial_name_prefix="SageMakerTrial"):
             experiment_name=self.experiment_name,
             sagemaker_boto_client=self.sagemaker_boto_client,
         )
+
+    def delete_all(self, action):
+        """
+        Force to delete the experiment and associated trials, trial components under the experiment.
+
+        Args:
+            action (str): pass in string 'force--' to confirm recursively delete all the experiments, trials,
+            and trial components.
+        """
+        if action != "force--":
+            raise ValueError(
+                "Must confirm with string 'force--' in order to delete the experiment and "
+                "associated trials, trial components."
+            )
+
+        for trial_summary in self.list_trials():
+            t = trial.Trial.load(sagemaker_boto_client=self.sagemaker_boto_client, trial_name=trial_summary.trial_name)
+            for trial_component_summary in t.list_trial_components():
+                tc = trial_component.TrialComponent.load(
+                    sagemaker_boto_client=self.sagemaker_boto_client,
+                    trial_component_name=trial_component_summary.trial_component_name,
+                )
+                t.remove_trial_component(tc)
+                tc.delete(disassociate=True)
```

Comment:
Is the first step needed given the second step will disassociate from all trials?

Reply:
thats true, fixed